erpnext/accounts/doctype/payment_entry/payment_entry.py

```diff
@@ -393,6 +393,10 @@ def set_unallocated_amount(self):
 			if self.payment_type == "Receive" \
 				and self.base_total_allocated_amount < self.base_received_amount + total_deductions \
 				and self.total_allocated_amount < self.paid_amount + (total_deductions / self.source_exchange_rate):
+					# Nonetype throws type error while performing mathematical manipulation thus assigned a value zero 
+					if self.base_total_discounted_amount is None:
+						self.base_total_discounted_amount = 0
+					
 					self.unallocated_amount = (self.base_received_amount + total_deductions - self.base_total_discounted_amount -
 						self.base_total_allocated_amount) / self.source_exchange_rate
```

Comment:
The test case was failing due to the mathematical calculation of the 
![testcase_sales_invoice_test_outstanding_amount_after_advance_payment_entry_cancelation](https://user-images.githubusercontent.com/80836439/130692777-f06ce661-226c-497a-bd7d-cb9e3e15c5a7.png)
NoneType operand. Please refer to below image.